pydantic/type_adapter.py

```diff
@@ -233,100 +208,124 @@ def __init__(
         self._type = type
         self._config = config
         self._parent_depth = _parent_depth
+
         if module is None:
             f = sys._getframe(1)
             self._module_name = cast(str, f.f_globals.get('__name__', ''))
         else:
             self._module_name = module
 
-        self._core_schema: CoreSchema | None = None
-        self._validator: SchemaValidator | PluggableSchemaValidator | None = None
-        self._serializer: SchemaSerializer | None = None
+        self.core_schema: CoreSchema
+        self.validator: SchemaValidator | PluggableSchemaValidator
+        self.serializer: SchemaSerializer
+
+        localns: _namespace_utils.MappingNamespace = (
+            _typing_extra.parent_frame_namespace(parent_depth=self._parent_depth) or {}
```

Comment:
Seems like the following test case fails:

https://github.com/pydantic/pydantic/blob/e9cde451034f4c9359a92ec7e4bddc0bfc0af0d1/tests/test_types_namedtuple.py#L199-L207

Because it can't find `T`, but surely this `localns` should include it, right?

Reply:
Resolving, we've fixed this specific issue 👍